src_c/pgcompat.h

```diff
@@ -223,3 +223,5 @@ SDL_Surface * SDL_CreateRGBSurfaceWithFormat(Uint32 flags, int width, int height
 #endif
 
 #endif /* ~PGCOMPAT_INTERNAL_H */
+
+#define WINDOWS_IGNORE_PACKING_MISMATCH
```

Comment:
What is this needed for? Do we need it for all files?

Reply:
It is only needed to compile with recent windows SDKs, and only with SDL1. And I believe it just turns off the warning inside the windows headers. You can read someone complainign about it here:

https://developercommunity.visualstudio.com/content/problem/739697/c2118-error-with-windows-kits-100183620-in-winnth.html

It's a bug/incompatibility between one of the windows headers and code compiled with non 8 byte aligned structs as far as I can tell:

https://stackoverflow.com/questions/56570681/static-assertion-failed-with-windows-headers-require-the-default-packing-option

There are several complaints of it generating warnings in old code across the internet and adding this define appears to be the easy fix out there. 

see:

https://github.com/microsoft/vcpkg/issues/8748



I believe it comes to us via SDL 1 as we don't seem to be doing anything fancy with structs and SDL 2 seems to work fine, so any modules that use SDL would need it. It's possible someone will come up with a better fix in the future but it may be that SDL 1 would need to be rebuilt without /Zp? It's not 100% clear. Anyway, far as I can tell SDL 1 was rubbing along OK with this mismatch before now so maybe ignoring the warning with this define is the best way.

Reply:
Should't that define be _inside_ the "include-guard" if-statement rather than outside it to avoid issues if the file is included twice for some reason?

Reply:
Thanks for the explanation :)

Reply:
> Should't that define be _inside_ the "include-guard" if-statement rather than outside it to avoid issues if the file is included twice for some reason?

Probably! Though I don't think it'll make much difference for a simple define like this. I'll move it.

Reply:
I moved it and added more details of the files that are currently affected by the problem. 

While it definitely only happens on our SDL1 build, I'm now not sure it is coming in from SDL1 so it may be possible to fix it on our side somehow. The three .c files currently affected are scrap.c, display.c and event.c.